src/main/java/io/mvnpm/esbuild/install/WebDepsInstaller.java

```diff
@@ -33,80 +28,86 @@ public final class WebDepsInstaller {
-            if (!packageNameAndRoot.isEmpty()) {
-                for (Map.Entry<String, Path> nameAndRoot : packageNameAndRoot.entrySet()) {
-                    final String packageName = nameAndRoot.getKey();
-                    final Path source = nameAndRoot.getValue();
-                    final Path target = nodeModulesDir.resolve(packageName);
-                    dirs.add(packageName);
-                    PathUtils.deleteRecursive(target);
-                    Files.createDirectories(target.getParent());
-                    PathUtils.safeMove(source, target);
-                    logger.log(Level.FINE, "installed package ''{0}''", packageName);
+                final Map<String, Path> packageNameAndRoot = JarInspector.findPackageNameAndRoot(dep.id(), extractDir,
+                        dep.type());
+                List<String> dirs = new ArrayList<>();
+                if (!packageNameAndRoot.isEmpty()) {
+                    for (Map.Entry<String, Path> nameAndRoot : packageNameAndRoot.entrySet()) {
+                        final String packageName = nameAndRoot.getKey();
+                        final Path source = nameAndRoot.getValue();
+                        final Path target = nodeModulesDir.resolve(packageName);
+                        dirs.add(packageName);
+                        PathUtils.deleteRecursive(target);
+                        Files.createDirectories(target.getParent());
+                        PathUtils.safeMove(source, target);
+                        logger.log(Level.FINE, "installed package ''{0}''", packageName);
+                    }
+                    installed.add(new MvnpmInfo.InstalledDependency(dep.id(), dirs));
+                    logger.log(Level.FINE, "installed dep ''{0}'' (''{1}'')", new Object[] { dep.path(), dep.id() });
+                } else {
+                    logger.log(Level.WARNING, "package.json not found in dep: ''{0}'' (''{1}'')",
+                            new Object[] { dep.path(), dep.id() });
                 }
-                installed.add(new MvnpmInfo.InstalledDependency(dep.id(), dirs));
-                logger.log(Level.FINE, "installed dep ''{0}'' (''{1}'')", new Object[] { dep.path(), dep.id() });
-            } else {
-                logger.log(Level.WARNING, "package.json not found in dep: ''{0}'' (''{1}'')",
-                        new Object[] { dep.path(), dep.id() });
+            } catch (IOException e) {
+                throw new RuntimeException(e);
             }
-        }
+        });
         PathUtils.deleteRecursive(tmp);
-        Set<String> installedDirs = installed.stream().flatMap(i -> i.dirs().stream()).collect(Collectors.toSet());
-        Set<String> legacyDirs = mvnpmInfo.installed().stream().flatMap(i -> i.dirs().stream())
-                .collect(Collectors.toSet());
-        // we are not deleting all the legacy dependencies, some of the dirs might still be used by new ones (e.g version or classifier change).
```

Comment:
@edewit I like your new algo which is much simpler, but it seems to be deleting all the dirs of legacy dependencies